packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -0,0 +1,551 @@
+                          )
+                        )
+                      )
+                    )
+                  ]
+                )
+              )
+            )
+          );
+        }
+
+        bottomNavigation = new SizedBox(
+          width: _maxWidth,
+          child: new Row(
+            children: children
+          )
+        );
+
+        break;
+    }
+
+    return new Stack(
+      children: <Widget>[
+        new Positioned.fill(
+          child: new Material(
+            elevation: 8,
+            color: _backgroundColor
+          )
+        ),
+        new SizedBox(
+          height: kBottomNavigationBarHeight,
+          child: new Center(
+            child: new Stack(
+              children: <Widget>[
+                new Positioned(
+                  left: 0.0,
+                  top: 0.0,
+                  right: 0.0,
+                  bottom: 0.0,
+                  child: new CustomPaint(
+                    painter: new _RadialPainter(
+                      circles: _circles.toList()
+                    )
+                  )
+                ),
+                new Material(
+                  type: MaterialType.transparency,
+                  child: new Center(
+                    child: bottomNavigation
+                  )
+                )
+              ]
+            )
+          )
+        )
+      ]
+    );
+  }
+}
+
+class _Circle {
+  _Circle({
+    this.state,
+    this.index,
+    this.color
+  }) {
+    controller = new AnimationController(
+      duration: kThemeAnimationDuration
+    );
+    animation = new CurvedAnimation(
+      parent: controller,
+      curve: Curves.fastOutSlowIn
+    );
+
+    controller.forward();
+  }
+
+  final _BottomNavigationBarState state;
+  final int index;
+  final Color color;
+  AnimationController controller;
+  CurvedAnimation animation;
+
+  FractionalOffset get offset {
+    return state.cirleOffset(index);
+  }
+
+  void dispose() {
+    controller.dispose();
+  }
+
+  @override
+  int get hashCode => hashValues(offset, controller, color);
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! _Circle)
+      return false;
+    final _Circle typedOther = other;
+    return offset == typedOther.offset &&
+           controller == typedOther.controller &&
+           color == typedOther.color;
+  }
+}
+
+class _RadialPainter extends CustomPainter {
+  _RadialPainter({
+    this.circles
+  });
+
+  final List<_Circle> circles;
+
+  double _maxRadius(FractionalOffset offset, Size size) {
+    final double dx = offset.dx;
+    final double dy = offset.dy;
+    final double x = (dx > 0.5 ? dx : 1.0 - dx) * size.width;
+    final double y = (dy > 0.5 ? dy : 1.0 - dy) * size.height;
+    return math.sqrt(x * x + y * y);
+  }
+
+  @override
+  bool shouldRepaint(_RadialPainter oldPainter) {
+    if (circles == oldPainter.circles)
```

Comment:
Maybe I'm missing something. Isn't this test and the following ones backwards? You should repaint when the lists of circles have changed, right?